src/bioutils/seqfetcher.py

```diff
@@ -139,11 +139,25 @@ def _fetch_seq_ensembl(ac, start_i=None, end_i=None):
         True
     """
 
-    url_fmt = "http://rest.ensembl.org/sequence/id/{ac}"
-    url = url_fmt.format(ac=ac)
+    # Ensembl API only takes transcript IDs (without version) and returns the latest one
+    # So we need to strip the transcript version, then check if what was returned was the one we requested
+    version = None
+    m = re.match(r"^(ENST\d+)\.(\d+)$", ac)
+    if m:
+        ac, version = m.groups()
+        version = int(version)
+
+    url = f"http://rest.ensembl.org/sequence/id/{ac}"
     r = requests.get(url, headers={"Content-Type": "application/json"})
     r.raise_for_status()
-    seq = r.json()["seq"]
+    data = r.json()
+    if version is not None:
+        latest_version = data["version"]
+        if version != latest_version:
+            msg = f"Ensembl API only provides {ac} version ({latest_version}), requested: {version}"
+            raise requests.RequestException(msg)
```

Comment:
@davmlaw : Can we please raise KeyError here instead? The error doesn't have anything to do with http or requests, so an application exception (rather than a protocol exception) seems more appropriate to me.

Sorry I didn't catch this earlier.